azurelinuxagent/ga/update.py

```diff
@@ -341,7 +342,17 @@ def run(self, debug=False):
                         op=WALAEventOperation.ProcessGoalState,
                         duration=duration,
                         message="Incarnation {0}".format(exthandlers_handler.last_etag))
-
+                if last_vm_artifacts_seqno != exthandlers_handler.last_vm_artifacts_seqno:
+                    self._ensure_readonly_files()
+                    duration = elapsed_milliseconds(utc_start)
```

Comment:
Apart from the Prefix - `Incarnation` vs `SeqNo`, everything from line 345 to 355 is the same as 335 to 344. 
In an effort to reduce code redundancy as much as we can, could you please make this into a small private/lambda function and pass the prefix as a parameter. Thanks!